### PR TITLE
All IBM: modifs in Connector, Generator, and Geom. Post.Rotor: minor

### PR DESCRIPTION
**Pull request by BconstantMMK**
Connector.IBM: remove expand=4 for frontType42 and add prepareIBMDataAdapt.

Connector.getIBMPtsWithFront: tolerance changes.

Geom.IBM: bug fix for closeSurface

Generator.IBM: addRefinementZones__ modifications to refine more cells within the refinement bodies. The blanking operation is now performed at the nodes. The adaptation process remains unchanged. A more robust solution will follow to deal with more complex configurations.

Post.Rotor: bug fix for extractSlices when sliceNature == 'curved'

--------------

The following test cases must be updated: FastIBMWireMeshModel_t1.py and IBMOutletControlPressure_t1.py
(This is due to the tolerance changes in Connector.getIBMPtsWithFront)

### CODE REVIEW

**Comment by antjost** [comment]
Provided by @BconstantMMK 
Examples of the previous and new behavior for Generator.IBM: addRefinementZones__

Previous behavior:: 


New behavior::

File: Cassiopee/Generator/Generator/IBM.py [hunk 1]
```diff
@@ -1120,7 +1132,7 @@ def createRefinementBodies(tb, dimPb=3, hmod=0.01):
     import Geom.IBM as D_IBM
     import Geom.Offset as O
 
-    pointsPerUnitLength = 10 if dimPb == 3 else 100
+    pointsPerUnitLength = 25 if dimPb == 3 else 1000
```
Comment (antjost) [inline Cassiopee/Generator/Generator/IBM.py:1135]: Could these eventually be made into user input variables with default values? It has been shown that the previously chosen good parameters have evolved as more test cases are tested.

File: Cassiopee/Connector/Connector/IBM.py [hunk 2]
```diff
@@ -227,8 +227,6 @@ def prepareIBMData(t_case, t_out, tc_out, t_in=None, to=None, tbox=None, tinit=N
         Reynolds = Internal.getValue(Reynolds)
         if Reynolds < 1.e5: frontType = 1
     else: Reynolds = 1.e6
-
-    if frontType == 42: expand= 4
```
Comment (antjost) [inline Cassiopee/Connector/Connector/IBM.py:231]: Does F42 still require expand 4 if the offset refinement is not used? If so, could this if statement be modified such that expand=4 is still set in the code if the offset refinements are not provided?

File: Cassiopee/Generator/Generator/IBM.py [hunk 3]
```diff
@@ -934,22 +940,28 @@ def addRefinementZones__(o, tb, tbox, snearsf, vmin, dim):
...
+            to = C.node2Center(to, 'cellN')
+            C._initVars(to, '{centers:cellN}=({centers:cellN}>0.9)')
+            Internal._rmNodesFromName(to, Internal.__FlowSolutionNodes__)
             fact = 1.1
             while C.getMinValue(to, 'centers:cellN') == 1 and fact < 10.:
                 print("Info: addRefinementZones: tbox too small - increase tbox by fact = %2.1f"%(fact))
                 box2 = T.scale(box, fact) 
                 tboxl[2][1][2] = box2
-                to = X_IBM.blankByIBCBodies(to, tboxl, 'centers', dim)
+                to = X_IBM.blankByIBCBodies(to, tboxl, 'centers', 3)
```
Comment (antjost) [inline Cassiopee/Generator/Generator/IBM.py:964]: is this a typo or is 3 also used for 2D problems?

**Review by antjost — COMMENTED** [review]
First time doing this. Not sure if I did it correctly. Let me know.

File: Cassiopee/Generator/Generator/IBM.py (hunk 3 above)
Comment (BconstantMMK) [reply Cassiopee/Generator/Generator/IBM.py:964]: It's not a typo, I use the node-centered blankCellsTri algorithm for both 2D and 3D cases! This also explains the addkplane at the beginning for 2D tboxes.

File: Cassiopee/Generator/Generator/IBM.py (hunk 1 above)
Comment (BconstantMMK) [reply Cassiopee/Generator/Generator/IBM.py]: Ah, sorry, you told me that last time and I forgot to change that bit! I modified the code in another commit.

**Comment by antjost** [comment]
👍

**Pull request merged**